service/task.go

```diff
@@ -640,6 +641,74 @@ func (uis *UIServer) taskLogRaw(w http.ResponseWriter, r *http.Request) {
 	uis.render.Stream(w, http.StatusOK, data, "base", "task_log.html")
 }
 
+func (uis *UIServer) taskFileRaw(w http.ResponseWriter, r *http.Request) {
+	projCtx := MustHaveProjectContext(r)
+	if projCtx.Task == nil {
+		http.Error(w, "task not found", http.StatusNotFound)
+		return
+	}
+
+	fileName := gimlet.GetVars(r)["file_name"]
+	if fileName == "" {
+		uis.LoggedError(w, r, http.StatusBadRequest, errors.New("file name not specified"))
+		return
+	}
+
+	taskFiles, err := artifact.GetAllArtifacts([]artifact.TaskIDAndExecution{{TaskID: projCtx.Task.Id, Execution: projCtx.Task.Execution}})
+	if err != nil {
+		uis.LoggedError(w, r, http.StatusInternalServerError, errors.Wrapf(err, "unable to find artifacts for task: %s", projCtx.Task.Id))
+	}
+	taskFiles, err = artifact.StripHiddenFiles(taskFiles, true)
+	if err != nil {
+		uis.LoggedError(w, r, http.StatusInternalServerError, errors.Wrapf(err, "unable to strip hidden files for task: %s", projCtx.Task.Id))
+	}
+	var tFile *artifact.File
+	for _, taskFile := range taskFiles {
+		if taskFile.Name == fileName {
+			tFile = &taskFile
+			break
+		}
+	}
+	if tFile == nil {
+		uis.LoggedError(w, r, http.StatusNotFound, errors.New("File not found"))
+		return
+	}
+	if !utility.StringMatchesAnyRegex(tFile.ContentType, uis.Settings.Ui.FileStreamingContentTypes) {
```

Comment:
Should this really match a regex? I imagine it's a small enough number that it doesn't super matter, but I would expect this to have been an exact string match.

Reply:
This allows us to support partial matches https://docs.google.com/document/d/10qJmO3lkMvUBpOApXu9ovy9m0ARBQGbaOvoFUbRW6FE/edit?disco=AAAA21_7eZc

Reply:
Ah, that makes sense. Would all these cases be prefix matches, in which case you could use `strings.HasPrefix()`?

Reply:
Done!